yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -56,10 +59,16 @@ private[spark] class YarnClientSchedulerBackend(
     totalExpectedExecutors = args.numExecutors
     client = new Client(args, conf)
     appId = client.submitApplication()
+    services.start(sc, appId)
+    services.addShutdownHook(ApplicationMaster.SHUTDOWN_HOOK_PRIORITY - 1)
```

Comment:
Do you still need to do the same thing in YarnClusterSchedulerBackend? or spark-1.4 consolidate then together?